contracts/SpokePool.sol

```diff
@@ -1621,6 +1667,118 @@ abstract contract SpokePool is
+            // were originally expecting. If it is negative, then the recipient will receive less.
+            // -1e18 is -100%. Because we cannot pay out negative values, that is the minimum.
+            // Moreover, allow the payout adjustment to go up to 1000% (i.e. 11x).
+            // This is a sanity check to ensure the payouts do not grow too large via some sort of issue in bundle
+            // construction.
+
+            if (payoutAdjustmentPct < -1e18 || payoutAdjustmentPct > 100e18) revert InvalidPayoutAdjustmentPct();
+
+            // Note: since _computeAmountPostFees is typically intended for fees, the signage must be reversed.
+            amountToSend = _computeAmountPostFees(amountToSend, -payoutAdjustmentPct);
+        }
+
+        // If relayer and receiver are the same address, there is no need to do any transfer, as it would result in no
+        // net movement of funds.
+        // Note: this is important because it means that relayers can intentionally self-relay in a capital efficient
+        // way (no need to have funds on the destination).
+        // If this is a slow fill, we can't exit early since we still need to send funds out of this contract
+        // since there is no "relayer".
+        bool isSlowFill = relayExecution.slowFill;
+        address recipientToSend = relayExecution.updatedRecipient;
+        if (msg.sender == recipientToSend && !isSlowFill) return;
+
+        // If relay token is wrappedNativeToken then unwrap and send native token.
+        address outputToken = relayData.outputToken;
+        if (outputToken == address(wrappedNativeToken)) {
+            // Note: useContractFunds is True if we want to send funds to the recipient directly out of this contract,
+            // otherwise we expect the caller to send funds to the recipient. If useContractFunds is True and the
+            // recipient wants wrappedNativeToken, then we can assume that wrappedNativeToken is already in the
+            // contract, otherwise we'll need the user to send wrappedNativeToken to this contract. Regardless, we'll
+            // need to unwrap it to native token before sending to the user.
+            if (!isSlowFill) IERC20Upgradeable(outputToken).safeTransferFrom(msg.sender, address(this), amountToSend);
+            _unwrapwrappedNativeTokenTo(payable(recipientToSend), amountToSend);
+            // Else, this is a normal ERC20 token. Send to recipient.
+        } else {
+            // Note: Similar to note above, send token directly from the contract to the user in the slow relay case.
+            if (!isSlowFill) IERC20Upgradeable(outputToken).safeTransferFrom(msg.sender, recipientToSend, amountToSend);
+            else IERC20Upgradeable(outputToken).safeTransfer(recipientToSend, amountToSend);
+        }
+
+        // @TODO Can we remove the third param in handleAcrossMessage since we no longer support partial fills? Will
```

Comment:
Ideally, yes. Maybe we should ask that integrators implement both, deprecate the old one and schedule a switch-over date at the launch of the USS. Thoughts?

Reply:
That's an interesting idea. @pxrl do you think that's something that could be useful? Also do we have an accurate idea of the total integrator set?

Reply:
My inclination is get this in soon with a breaking change, since we only have a few integrators at the moment. Ideally we can coordinate this with them so that they're ready to update when the new interface goes live. I'll check with the integrators directly.